components/sup/src/event.rs

```diff
@@ -69,29 +72,15 @@ pub fn init_stream(conn_info: EventConnectionInfo, event_core: EventCore) {
         });
 }
 
-/// Publish an event. This is the main interface that client code will
-/// use.
-///
-/// If `init_stream` has not been called already, this function will
-/// be a no-op.
-// NOTE: we can take advantage of this to "disable" the event
-// subsystem if users don't wish to send events out; just don't call
-// `init_stream` if they don't want it.
-pub fn publish(event: &impl Event) {
-    // TODO: incorporate the current timestamp into the rendered event
-    // (which will require tweaks to the rendering logic, but we know
-    // that'll need to be updated anyway).
-
-    // We render the event to bytes here, rather than over in the
-    // publication thread, because it allows our Event types to deal
-    // with references, which means less allocations and unnecessary
-    // copying (otherwise, we'd have to have copies of everything to
-    // send it over to the other thread, or much more complicated
-    // thread-safe types!). It also makes the publication thread logic
-    // a bit simpler; it just takes bytes and sends them out.
-    if let Some(e) = EVENT_STREAM.try_get::<EventStream>() {
-        e.send(event.render(EVENT_CORE.get::<EventCore>()));
-    }
+/// All the information needed to establish a connection to a NATS
+/// Streaming server.
+// TODO: This will change as we firm up what the interaction between
+// Habitat and A2 looks like.
+pub struct EventConnectionInfo {
+    pub name:        String,
+    pub verbose:     bool,
+    pub cluster_uri: String,
+    pub cluster_id:  String,
```

Comment:
TODO: It is only missing the token to authenticate to the new A2 NATs Server living in the `event-gateway`. Just a thought to add in the future.

Reply:
depending on the rust NATS library, it might be possible to encode in the URL, like `nats://TOKEN@host.example:4222` (this is how the golang one works).